File: poh4/poh4_3.py
#!/usr/bin/python
import sys
if sys.version_info[0]>=3: raw_input=input

T,N=[int(e) for e in raw_input().split()]
a=[0]*N
cur=i=0
while i<T:
	a[i]=int(raw_input())
	cur+=a[i]
	i+=1
r=cur
while i<N:
	a[i]=int(raw_input())
	cur+=a[i]
	cur-=a[i-T]
	if r<cur: r=cur
	i+=1
print(r)

File: poh4/poh4_3.php
#!/usr/bin/php
<?php
list($T,$N)=array_map(intval,split(" ",fgets(STDIN)));
$a=array();
$cur=0;
$i=0;
for(;$i<$T;$i++){
	$a[$i]=intval(fgets(STDIN));
	$cur+=$a[$i];
}
$r=$cur;
for(;$i<$N;$i++){
	$a[$i]=intval(fgets(STDIN));
	$cur+=$a[$i];
	$cur-=$a[$i-$T];
	if($r<$cur)$r=$cur;
}
echo $r.PHP_EOL;